src/performance/monitor.py

```
@@ -0,0 +1,132 @@
+
+import time
+import psutil
+from datetime import datetime
+from typing import Dict, List, Any
+from collections import deque
+
+class PerformanceMonitor:
+    def __init__(self, history_size: int = 1000):
+        self.metrics = {}
+        self.history = deque(maxlen=history_size)
+        self.start_time = datetime.now()
```

Ruff (0.8.2)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)